From: Steve McIntyre <93sam@debian.org>
Date: Fri, 13 Apr 2018 18:21:09 +0100
Subject: [13/31] drivers: net: generalize napi_complete_done()
Origin: https://git.kernel.org/linus/6ad20165d376fa07919a70e4f43dfae564601829

napi_complete_done() allows to opt-in for gro_flush_timeout, added
back in linux-3.19, commit 3b47d30396ba ("net: gro: add a per device
gro flush timer")

This allows for more efficient GRO aggregation without sacrifying
latencies.

Signed-off-by: Eric Dumazet <edumazet@google.com>
Signed-off-by: David S. Miller <davem@davemloft.net>

This is just a trivial extraction of the emac-specific parts of commit
6ad20165d376fa07919a70e4f43dfae564601829
---
 drivers/net/ethernet/qualcomm/emac/emac.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/qualcomm/emac/emac.c b/drivers/net/ethernet/qualcomm/emac/emac.c
index 27cfa5ff51a5..34090f7ed833 100644
--- a/drivers/net/ethernet/qualcomm/emac/emac.c
+++ b/drivers/net/ethernet/qualcomm/emac/emac.c
@@ -129,7 +129,7 @@ static int emac_napi_rtx(struct napi_struct *napi, int budget)
 	emac_mac_rx_process(adpt, rx_q, &work_done, budget);
 
 	if (work_done < budget) {
-		napi_complete(napi);
+		napi_complete_done(napi, work_done);
 
 		irq->mask |= rx_q->intr;
 		writel(irq->mask, adpt->base + EMAC_INT_MASK);
-- 
2.11.0

